meter/goodwe-wifi.go

```diff
@@ -0,0 +1,172 @@
+package meter
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+)
+
+type Server struct {
+	conn      *net.UDPConn
+	inverters map[string]Inverter
+}
+
+type Inverter struct {
+	IP           string
+	pvPower      float64
+	netPower     float64
+	batteryPower float64
+	soc          float64
+}
+
+var server *Server
+
+type GoodWeWiFiMeter struct {
+	usage string
+	URI   string
+}
+
+func init() {
+	registry.Add("goodwe-wifi", NewGoodWeWifiFromConfig)
+}
+
+//
+//go:generate go run ../cmd/tools/decorate.go -f decorateBoschBpts5Hybrid -b api.Meter -t "api.Battery,Soc,func() (float64, error)"
+func NewGoodWeWifiFromConfig(other map[string]interface{}) (api.Meter, error) {
+	cc := struct {
+		capacity   `mapstructure:",squash"`
+		URI, Usage string
+		Cache      time.Duration
+	}{
+		Cache: time.Second,
+	}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	return NewGoodWeWiFiMeter(cc.URI, cc.Usage)
+}
+
+func NewGoodWeWiFiMeter(uri string, usage string) (api.Meter, error) {
+	meter := &GoodWeWiFiMeter{
+		usage: usage,
+		URI:   uri,
+	}
+
+	server, err := NewServer()
```

Comment:
Wie funktioniert das bei >1 Inverter?

Reply:
Funktioniert, habs bei mir mit 2 Stück laufen.

Deshalb prüfe ich in Zeile 80 obs den Server schon gibt.